crates/rtmp/src/lib.rs

```diff
@@ -1,16 +1,380 @@
-mod channels;
+//! A crate for handling RTMP server connections.
+//!
+//! ## Specifications
+//!
+//! | Name | Version | Link | Comments |
+//! | --- | --- | --- | --- |
+//! | Video File Format Specification | `10` | <https://github.com/veovera/enhanced-rtmp/blob/main/docs/legacy/video-file-format-v10-0-spec.pdf> | |
+//! | Adobe Flash Video File Format Specification | `10.1` | <https://github.com/veovera/enhanced-rtmp/blob/main/docs/legacy/video-file-format-v10-1-spec.pdf> | Refered to as 'Legacy FLV spec' in this documentation |
+//! | Enhancing RTMP, FLV | `v1-2024-02-29-r1` | <https://github.com/veovera/enhanced-rtmp/blob/main/docs/enhanced/enhanced-rtmp-v1.pdf> | |
+//! | Enhanced RTMP | `v2-2024-10-22-b1` | <https://github.com/veovera/enhanced-rtmp/blob/main/docs/enhanced/enhanced-rtmp-v2.pdf> | Refered to as 'Enhanced RTMP spec' in this documentation |
+//!
+//! ## Status
+//!
+//! This crate is currently under development and is not yet stable.
+//!
+//! Unit tests are not yet fully implemented. Use at your own risk.
+//!
+//! ## License
+//!
+//! This project is licensed under the [MIT](./LICENSE.MIT) or [Apache-2.0](./LICENSE.Apache-2.0) license.
+//! You can choose between one of them if you use this work.
+//!
+//! `SPDX-License-Identifier: MIT OR Apache-2.0`
+#![cfg_attr(all(coverage_nightly, test), feature(coverage_attribute))]
+#![deny(missing_docs)]
+#![deny(unsafe_code)]
+
 mod chunk;
+mod command_messages;
+mod error;
 mod handshake;
-mod macros;
 mod messages;
-mod netconnection;
-mod netstream;
 mod protocol_control_messages;
 mod session;
 mod user_control_messages;
 
-pub use channels::{ChannelData, DataConsumer, DataProducer, PublishConsumer, PublishProducer, PublishRequest, UniqueID};
-pub use session::{Session, SessionError};
+pub use session::{Session, SessionData, SessionError, SessionHandler};
```

Comment:
We have some mistakes here, we export `SessionError` but this doesnt rexport the sub error type.s